src/component-library/Tabs/Tab.tsx

```diff
@@ -0,0 +1,23 @@
+import { useTab } from '@react-aria/tabs';
+import { TabListState } from '@react-stately/tabs';
+import { AriaTabProps } from '@react-types/tabs';
+import { ReactNode, useRef } from 'react';
+
+import { StyledTab } from './Tabs.style';
+
+type TabProps<T> = { item: AriaTabProps & { rendered: ReactNode }; state: TabListState<T> };
+
+// @internal
```

Comment:
The reason I add this `@internal` is to clarify that this component is no supposed to use outside of Tabs component.